lib/index.ts

```diff
@@ -29,31 +29,40 @@ export const generateRemoteUrl = (service: string) => window.location.protocol +
 /**
  * Get the base path for the given OCS API service
  *
- * @param {string} service name
- * @param {int} version OCS API version
- * @return {string} OCS API base path
+ * @param {string} url OCS API service url
+ * @param {object} params parameters to be replaced into the service url
+ * @param {UrlOptions} options options for the parameter replacement
+ * @return {string} Absolute path for the OCS URL
  */
-export const generateOcsUrl = (service: string, version: Number) => {
-    version = (version !== 2) ? 1 : 2
-    return window.location.protocol + '//' + window.location.host + getRootUrl() + '/ocs/v' + version + '.php/' + service + '/'
+export const generateOcsUrl = (url: string, params?: object, options?: UrlOptions) => {
+    const allOptions = Object.assign({
+        ocsVersion: 2
```

Comment:
can we somehow add the ``ocsVersion``option to the jsdoc?

Reply:
A doc block on `export interface UrlOptions {` is enough? Or should I doc it on this function directly?

Reply:
I would document `options.ocsVersion` on the function, I think that works with more editors/IDEs

Reply:
Done